Src/PCompiler/CompilerCore/Backend/Stately/StatelyCodeGenerator.cs

```diff
@@ -0,0 +1,217 @@
+                    {
+                        foreach (var stmt in ifStmt.ElseBranch.Statements)
+                        {
+                            gotoStmts.AddRange(WriteStmt(stmt));;
+                        }
+                    }
+                    break;
+                case WhileStmt whileStmt:
+                    foreach (var stmt in whileStmt.Body.Statements)
+                    {
+                        gotoStmts.AddRange(WriteStmt(stmt));
+                    }
+                    break;
+                case ForeachStmt foreachStmt:
+                    foreach (var stmt in foreachStmt.Body.Statements)
+                    {
+                        gotoStmts.AddRange(WriteStmt(stmt));
+                    }
+                    break;
+                case CompoundStmt compoundStmt:
+                    foreach (var stmt in compoundStmt.Statements)
+                    {
+                        gotoStmts.AddRange(WriteStmt(stmt));
+                    }
+                    break;
+                case ReceiveStmt receiveStmt:
+                    foreach (var (key, value) in receiveStmt.Cases)
+                    {
+                        foreach (var stmt in value.Body.Statements)
+                        {
+                            gotoStmts.AddRange(WriteStmt(stmt));
+                        }
+                    }
+                    break;
+            }
+
+            return gotoStmts;
+        }
+        
+        //Handles writing all instances of a machine
+        private void WriteMachine(CompilationContext context, StringWriter output, Machine machine)
+        {
+
+            context.WriteLine(output, $"const {machine.Name} = createMachine<Context>({{");
+            context.WriteLine(output, $"id: \"{machine.Name}\",");
+            
+            //Code start state of machine.
+            context.WriteLine(output,$"initial: \"{machine.StartState.Name}\", ");
+            
+            //Code up the states in each machine.
+            context.WriteLine(output, "states: {");
+            foreach (State state in machine.States)
+            {
+                
+                context.WriteLine(output,$"{state.Name}: {{");
+                WriteState(context, output, state);
+                context.WriteLine(output, state.Equals(machine.States.Last()) ? "}" : "},");
+            }
+            context.WriteLine(output, "}");
+            context.WriteLine(output, "});");
+        }
+        
+        //Handles writing all instances of states (within a machine)
+        private void WriteState(CompilationContext context, StringWriter output, State state)
+        {
+            var entryGotoStmts = new List<String>();
+            //Entry function exists!
+            if (state.Entry != null) {
+                foreach (var s in state.Entry.Body.Statements) {
+                    entryGotoStmts.AddRange(WriteStmt(s));
+                }
+
+                if (entryGotoStmts.Count > 0)
```

Comment:
how does this work for the following case:

```
entry {
if(x > 0)
goto S1;
else
goto S2;
} 
```

Where does the `always` point to?
```

Reply:
It will point to both. There will be 2 always squares, one pointing to S1 and one pointing S2.